src/idl_gen_js.cpp

```diff
@@ -110,16 +144,23 @@ class JsGenerator : public BaseGenerator {
   std::string &exports = *exports_ptr;
   for (auto it = sorted_namespaces.begin();
        it != sorted_namespaces.end(); it++) {
-    code += "/**\n * @const\n * @namespace\n */\n";
-    if (it->find('.') == std::string::npos) {
-      code += "var ";
-      if(parser_.opts.use_goog_js_export_format) {
-        exports += "goog.exportSymbol('" + *it + "', " + *it + ");\n";
-      } else {
-        exports += "this." + *it + " = " + *it + ";\n";
+    if (ts_) {
+      if (it->find('.') == std::string::npos) {
+        code += "import { flatbuffers } from \"./flatbuffers\"\n";
```

Comment:
I see. I suppose either shipping `flatbuffers.ts`, or a `flatbuffers.d.ts` is acceptable. The latter is probably superior since it would avoid code duplication between the javascript & the typescript versions?